public/js/snackbar.js

```diff
@@ -39,35 +39,27 @@ class Snackbar {
      * @param {timeoutCallback} options.timeoutFunction - What to run on timeout (doesn't run if hidden or destroyed)
      * @param {string} options.timeoutMode - can be "destroy", "hide", "none" or empty. Determines what to do on timeout, destroys by default
      */
-    constructor(text, options = {destroyWhenButtonClicked: true, destroyWhenBodyClicked: true}) {
+    constructor(text, options = {destroyWhenButtonClicked: true, destroyWhenBodyClicked: true, timeoutFunction: () => {}, timeoutEndFunction: "destroy" }) {
         debugger;
         this.text = text;
-        this.color = options.color;
-        this.textColor = options.textColor;
-        this.buttonText = options.buttonText;
-        this.buttonClick = options.buttonClick;
-        this.destroyWhenButtonClicked = options.destroyWhenButtonClicked;
-        this.bodyClick = options.bodyClick;
-        this.destroyWhenBodyClicked = options.destroyWhenBodyClicked;
+        Object.assign(this, options);
 
         //timeout logic
-        this.timeoutFunction = options["timeoutFunction"] !== undefined ? options["timeoutFunction"] : () => {};
-        this.timeout = options["timeout"];
         this.timeoutInProgress;
 
         //what to run on timeout
         this.timeoutEndFunction;
-        switch(options["timeoutMode"]) {
-            case "destroy":
-            case undefined:
-                this.timeoutEndFunction = () => this.destroy();
-                break;
+        switch(options.timeoutMode) {
```

Comment:
We should use a space after `switch` because it is a keyword and not a function (in the same vein as `if`, `for`, `while`, etc.)
```suggestion
        switch (options.timeoutMode) {
```